download-temporal-cli.js

```diff
@@ -0,0 +1,88 @@
+import { basename, join } from 'path';
+import { createReadStream, createWriteStream } from 'fs';
+import { chmod, unlink } from 'fs/promises';
+import { Readable } from 'stream';
+import { finished } from 'stream/promises';
+
+import zlib from 'zlib';
+import tar from 'tar-fs';
+import mkdirp from 'mkdirp';
+import rimraf from 'rimraf';
+import kleur from 'kleur';
+
+const reportError = (error, exitCode = 2, callback = () => {}) => {
+  console.error(kleur.bgRed('Error:'), kleur.red(error));
+  callback();
+  process.exit(2);
+};
+
+const destinationDirectory = './bin';
+const destination = join(destinationDirectory, 'cli');
+
+const latestRelease = await fetch(
+  'https://api.github.com/repos/temporalio/cli/releases/latest',
```

Comment:
in case this hits github rate limit we can swap with https://github.com/temporalio/cloudflare-release-manager#urls

Reply:
might make few changes though so temporal.download returns the archive itself instead of just a download link

Reply:
Yea, I think we should _totally_ do that but maybe I wait until you're done making breaking changes? Ideally, we shouldn't be running `pnpm install` too often anyway.

Reply:
You should use our temporal.download CDN.
In TS we've been bit by GH rate limits before.